File: Backend/DTOs/RequestedCourseGroupDto.cs
namespace Backend.DTOs
{
    /// <summary>A data transfer object for a requested course.</summary>
    public class RequestedCourseGroupDto
    {
        public Guid Id { get; set; }

        public ICollection<RequestedCourseDto> RequestedCourses { get; set; }
        public ExemptedCourseDto RequestedExemptedCourse { get; set; }
    }
}

File: Frontend/src/dtos/requested-course-group.dto.ts
import {GUID} from "../utils/guid";
import {RequestedCourseDto} from "./requested-course.dto";
import {ExemptedCourseDto} from "./exempted-course.dto";

export class RequestedCourseGroupDto {
  id: GUID;
  requestedCourses: ArrayLike<RequestedCourseDto>;
  requestedExemptedCourse: ExemptedCourseDto;
}
